core/overdrive.py

```diff
@@ -1267,22 +1268,32 @@ def book_info_to_metadata(
 
                 # Samples become links.
                 if "samples" in format:
-                    overdrive_name = format["id"]
-                    internal_names = list(cls.internal_formats(overdrive_name))
-                    if not internal_names:
-                        # Useless to us.
-                        continue
-                    for content_type, drm_scheme in internal_names:
-                        if Representation.is_media_type(content_type):
-                            for sample_info in format["samples"]:
-                                href = sample_info["url"]
+                    for sample_info in format["samples"]:
+                        href = sample_info["url"]
+                        # Have we already parsed this sample? Overdrive repeats samples per format
+                        if href in sample_hrefs:
+                            continue
+
+                        # Every sample has its own format type
+                        overdrive_name = sample_info.get("formatType")
```

Comment:
`overdrive_format_name`? Not a big deal, since I think it's used only in this short block.

Reply:
Better name though, updated